presence/rust/engine/src/ffi.rs

```diff
@@ -0,0 +1,105 @@
+use libc::size_t;
+use std::ptr;
+
+/// Struct to hold an action, identity type and their associated discovery condition.
+#[derive(Clone, Copy, Debug)]
+#[repr(C)]
+pub struct DiscoveryCondition {
+    pub action: i32,
+    pub identity_type: i32,
+    pub measurement_accuracy: i32,
```

Comment:
Not sure if we should use Enums instead?

Reply:
Good idea. Changed to enums.